src/windows/list.cpp

```diff
@@ -486,55 +512,76 @@ bool GetDetailData(DeviceDescriptor* device,
 
     GetMountpoints(deviceNumber, &device->mountpoints);
 
+    // printf("[INFO] Opening handle to %s\n", device->raw.c_str());
+
     hPhysical = CreateFileA(
-      device->raw.c_str(), handleOpenFlags,
+      device->raw.c_str(), 0,
       FILE_SHARE_READ | FILE_SHARE_WRITE, NULL,
       OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL);
 
+    // printf("[INFO] Opened handle to %s\n", device->raw.c_str());
+    // printf("[INFO] Handle %i (INVALID %i)\n",
+    //   hPhysical, INVALID_HANDLE_VALUE);
+
     if (hPhysical == INVALID_HANDLE_VALUE) {
       errorCode = GetLastError();
+      // printf("[INFO] Couldn't open handle to %s: Error %i\n",
+      //   device->raw.c_str(), errorCode);
       device->error = "Couldn't open handle to physical device: Error " +
         std::to_string(errorCode);
       result = false;
       break;
     }
 
+    // printf("[INFO] GetDeviceSize( %s )\n", device->raw.c_str());
+
     if (!GetDeviceSize(hPhysical, device)) {
       errorCode = GetLastError();
+      // printf("[ERROR] Couldn't get disk geometry: Error %i\n", errorCode);
       device->error = "Couldn't get disk geometry: Error " +
         std::to_string(errorCode);
       result = false;
       break;
     }
 
+    // printf("[INFO] GetAdapterInfo( %s )\n", device->raw.c_str());
+
     if (!GetAdapterInfo(hPhysical, device)) {
       errorCode = GetLastError();
+      // printf("[ERROR] Couldn't get device adapter descriptor: Error %i\n",
+      //   errorCode);
       device->error = "Couldn't get device adapter descriptor: Error " +
         std::to_string(errorCode);
       result = false;
       break;
     }
 
+    // printf("[INFO] GetDeviceBlockSize( %s )\n", device->raw.c_str());
+
     // NOTE: No need to fail over this one,
     // as we can safely default to a 512B block size
     if (!GetDeviceBlockSize(hPhysical, device)) {
       errorCode = GetLastError();
       // printf("[INFO] Couldn't get block size: Error %u\n", errorCode);
     }
 
+    // printf("[INFO] isWritable( %s )\n", device->raw.c_str());
+
     BOOL isWritable = DeviceIoControl(
       hPhysical, IOCTL_DISK_IS_WRITABLE, NULL, 0,
       NULL, 0, &size, NULL);
 
     device->isReadOnly = !isWritable;
-  }
+  }  // end for (index = 0; ; index++)
 
   if (hDevice != INVALID_HANDLE_VALUE) {
     CloseHandle(hDevice);
+    hDevice = INVALID_HANDLE_VALUE;
   }
 
   if (hPhysical != INVALID_HANDLE_VALUE) {
     CloseHandle(hPhysical);
+    hPhysical = INVALID_HANDLE_VALUE;
```

Comment:
Given the amount of places you do this, would it be worth creating an inline function wrapper (or macro) for this?